bazel/test/BUILD.bazel

```diff
@@ -0,0 +1,57 @@
+load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_test")
+load("@npm//@bazel/concatjs:index.bzl", "ts_library")
+load("//bazel:extract_types.bzl", "extract_types")
+load("//bazel:extract_js_module_output.bzl", "extract_js_module_output")
+
+ts_library(
+    name = "transitive_lib",
+    testonly = True,
+    srcs = ["transitive_file.ts"],
+)
+
+ts_library(
+    name = "test_lib",
+    testonly = True,
+    srcs = ["fixture.ts"],
+    deps = [":transitive_lib"],
+)
+
+filegroup(
+    name = "text_fixture",
+    testonly = True,
+    srcs = ["fixture.txt"],
+)
+
+extract_types(
+    name = "extract_types_target",
+    testonly = True,
+    deps = [":test_lib"],
+)
+
+extract_js_module_output(
+    name = "extract_js_module_output_target",
+    testonly = True,
+    forward_linker_mappings = True,
+    include_declarations = True,
+    include_default_files = True,
+    include_external_npm_packages = True,
+    provider = "JSModuleInfo",
```

Comment:
It's worth noting that we won't capture all possible code-paths here (e.g. all combinations of the target definitions), but that's out-of-scope and we are mostly interesting in the provider validation/analysis phase of Bazel